### DISCUSSION TITLE
How can I set a toolbar to stay at the top of the inside of a dock?

### DISCUSSION THREAD

**Discussion by i-make-robots** [andrewauclair/ModernDocking#158] [Q&A] [answered]
![image](https://github.com/andrewauclair/ModernDocking/assets/1464454/0ee8847e-acdc-4ef0-9e9b-761af04273fb)

Add Node and Remove Node are in a JToolBar on the north edge of the JPanel contained in the dock.  I'd like the toolbar to remain visible while the rest of the panel scrolls.  

Please tell me how?

**Comment by andrewauclair** [comment, marked as answer]
Wrote a new panel in the demo app as an example for you, https://github.com/andrewauclair/ModernDocking/blob/a78df1dd5ed06a805577d2b006c5c6a583ba009a/demo-single-app/src/basic/ScrollingWithToolbarPanel.java.

**Comment by i-make-robots** [comment]
Nice! It works great and now another panel refuses to fill the available width.  I am so dumb with Swing...

![image](https://github.com/andrewauclair/ModernDocking/assets/1464454/38ce18e4-f442-4f44-a32c-1ebd3803a745)

**Reply by i-make-robots** [reply]
I shouldn't see the box around "Pose".  I added a scrollpane but I'm not seeing it...

**Reply by i-make-robots** [reply]
file with the wonky look is https://github.com/MarginallyClever/Robot-Overlord-App/blob/ro3/src/main/java/com/marginallyclever/ro3/node/NodeDetailView.java

**Reply by i-make-robots** [reply]
Ha!  Found it.  JPanel does not default to BorderLayout...
